master/internal/stream/system_intg_test.go

```diff
@@ -465,6 +507,160 @@ func TestCheckpointStartup(t *testing.T) {
 	runStartupTest(t, testCases)
 }
 
+func TestMetricInsert(t *testing.T) {
+	pgDB, dbCleanup := db.MustResolveNewPostgresDatabase(t)
+	t.Cleanup(dbCleanup)
+
+	type metric struct {
```

Comment:
non-blocking: my only thought here is that if we need to test metrics in the future, wed want to move this out of this function. Im ok with this as it stands though and we can leave it to a future refactor

Reply:
Fair point! It'll be a good addition to `test_util.go`